working/augmentation-libraries/feature-specification.md

```diff
@@ -321,82 +322,150 @@ doesn't, it must repeat the type from the augmented definition__.
+    is the implicitly induced setter of the variable declaration.
+    _This also applies to `late` and `late final` variable declarations._
+    Type inference of `augmented = e1` is performed as a setter invocation
+    with the setter signature of the augmented declaration.
 
 *   **Augmenting functions**: When augmenting a function, `augmented`
-    refers to the augmented function. Tear offs are not allowed, so this
-    function must immediately be invoked.
-
-*   **Augmenting operators**: When augmenting an operator, `augmented`
-    refers to the augmented operator method, which must be immediately
-    invoked using function call syntax. For example, when augmenting
+    in the function body refers to the augmented function, and it must
+    occur only as the function of a function invocation, which means
+    it is immediately followed by one, or no, type argument list
+    and then an argument list.
+    _Tear offs are not possible, neither are member accesses like
+    `augmented.call()`._
+    The expression `augmented` must not occur in an optional parameter
+    default value expressions _It is a reserved word in such expressions,
+    but is currently cannot be used._
+    The term <code>'augmented' \<argumentPart\></code> words parses
+    as a `<primary>` expression.
+    Evaluating the function invocation, after having evaluated the
+    argument list, invokes the augmented function with the provided
+    type arguments and evaluated argument list, and evaluates to the
+    returned value.
+    Type inference of <Code>augmented\<_typeArgs_\>(_args_)<code>
+    is performed as a function invocation with the function signature
+    of the augmented declaration.
+
+*   **Augmenting operators**: Operators are methods with special
+    invocation syntax. When augmenting an operator declaration, `augmented`
+    refers to the augmented operator *method*, and it must occur only
+    as the function of a function invocation. For example, when augmenting
     `operator +` you could use `augmented(1)` to call the augmented
-    operator, and when augmenting `operator []=` you would use the
-    `augmented(key, value)` syntax.
-
-    *   Note that `augmented` in such an augmenting operator method body is
-        not an expression by itself, and cannot be used to tear off the
-        augmented operator method. Similar to `super`, it is a syntactic
-        form which can only be used in limited ways.
+    operator, when augmenting `operator []=` you would use the
+    `augmented(key, value)` syntax, and when augmenting `operator -()`,
+    you would use `augmented()`.
+    Parsing, evaluation and type inference is handled like a
+    function augmentation.
 
 *   **Augmenting enum values**: When augmenting an enum value, `augmented`
-    has no meaning and is not allowed.
+    has no meaning and is not allowed to occur in constructor argument
+    expressions.
+
+*   **Augmenting constructors**:
```

Comment:
I only got to here so far, will continue reviewing tomorrow (and I know this section will require more brain power than I currently have available 🤣 )